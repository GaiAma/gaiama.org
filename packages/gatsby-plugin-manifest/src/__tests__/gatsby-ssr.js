jest.mock(`fs`, () => {
  return {
    readFileSync: jest.fn().mockImplementation(() => `someIconImage`),
  }
})

jest.mock(`gatsby/dist/utils/create-content-digest`, () =>
  jest.fn(() => `contentDigest`)
)

const { onRenderBody } = require(`../gatsby-ssr`)

let headComponents
const setHeadComponents = args => (headComponents = headComponents.concat(args))

const ssrArgs = {
  setHeadComponents,
  pathname: `/`,
}

describe(`gatsby-plugin-manifest`, () => {
  beforeEach(() => {
    global.__PATH_PREFIX__ = ``
    headComponents = []
  })

  it(`Creates href attributes using pathPrefix`, () => {
    global.__PATH_PREFIX__ = `/path-prefix`

    onRenderBody(ssrArgs, {
      icon: true,
      theme_color: `#000000`,
    })

    headComponents
      .filter(component => component.type === `link`)
      .forEach(component => {
        expect(component.props.href).toEqual(
          expect.stringMatching(/^\/path-prefix\//)
        )
      })
  })

  describe(`Manifest Link Generation`, () => {
    it(`Adds a "theme color" meta tag to head if "theme_color_in_head" is not provided`, () => {
      onRenderBody(ssrArgs, {
        theme_color: `#000000`,
        include_favicon: false,
        cache_busting_mode: false,
        legacy: false,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Add a "theme color" meta tag if "theme_color_in_head" is set to true`, () => {
      onRenderBody(ssrArgs, {
        theme_color: `#000000`,
        theme_color_in_head: true,
        include_favicon: false,
        cache_busting_mode: false,
        legacy: false,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does not add a "theme color" meta tag if "theme_color_in_head" is set to false`, () => {
      onRenderBody(ssrArgs, {
        theme_color: `#000000`,
        theme_color_in_head: false,
        include_favicon: false,
        cache_busting_mode: false,
        legacy: false,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does not add a "theme_color" meta tag to head if "theme_color" option is not provided.`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        include_favicon: false,
        cache_busting_mode: false,
        legacy: false,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Adds correct i18n "manifest" link to head`, () => {
      const args = {
        ...ssrArgs,
        pathname: `/es/sobre-nosotros`,
      }
      onRenderBody(args, {
        manifests: [
          {
            start_url: `/de/`,
            regex: `^/de/.*`,
            language: `de`,
          },
          {
            start_url: `/en/`,
            regex: `^/en/.*`,
            language: `en`,
          },
          {
            start_url: `/es/`,
            regex: `^/es/.*`,
            language: `es`,
          },
        ],
      })
      expect(headComponents).toMatchSnapshot()
    })
  })

  describe(`Legacy Icons`, () => {
    describe(`Does create legacy links`, () => {
      it(`if "legacy" not specified in automatic mode`, () => {
        onRenderBody(ssrArgs, {
          icon: true,
          theme_color: `#000000`,
          include_favicon: false,
          cache_busting_mode: `none`,
          theme_color_in_head: false,
        })
        expect(headComponents).toMatchSnapshot()
      })

      it(`if "legacy" not specified in hybrid mode.`, () => {
        onRenderBody(ssrArgs, {
          icon: true,
          theme_color: `#000000`,
          icons: [
            {
              src: `/favicons/android-chrome-48x48.png`,
              sizes: `48x48`,
              type: `image/png`,
            },
            {
              src: `/favicons/android-chrome-512x512.png`,
              sizes: `512x512`,
              type: `image/png`,
            },
          ],
          include_favicon: false,
          cache_busting_mode: `none`,
          theme_color_in_head: false,
        })
        expect(headComponents).toMatchSnapshot()
      })

      it(`if "legacy" not specified in manual mode.`, () => {
        onRenderBody(ssrArgs, {
          icon: false,
          icons: [
            {
              src: `/favicons/android-chrome-48x48.png`,
              sizes: `48x48`,
              type: `image/png`,
            },
            {
              src: `/favicons/android-chrome-512x512.png`,
              sizes: `512x512`,
              type: `image/png`,
            },
          ],
        })
        expect(headComponents).toMatchSnapshot()
      })
    })

    describe(`Does not create legacy links`, () => {
      it(`If "legacy" options is false and in automatic`, () => {
        onRenderBody(ssrArgs, {
          icon: true,
          legacy: false,
          include_favicon: false,
          cache_busting_mode: false,
        })
        expect(headComponents).toMatchSnapshot()
      })

      it(`If "legacy" options is false and in manual mode`, () => {
        onRenderBody(ssrArgs, {
          icon: false,
          theme_color: `#000000`,
          legacy: false,
          icons: [
            {
              src: `/favicons/android-chrome-48x48.png`,
              sizes: `48x48`,
              type: `image/png`,
            },
            {
              src: `/favicons/android-chrome-512x512.png`,
              sizes: `512x512`,
              type: `image/png`,
            },
          ],
        })
        expect(headComponents).toMatchSnapshot()
      })

      it(`If "legacy" options is false and in hybrid mode`, () => {
        onRenderBody(ssrArgs, {
          icon: true,
          legacy: false,
          icons: [
            {
              src: `/favicons/android-chrome-48x48.png`,
              sizes: `48x48`,
              type: `image/png`,
            },
            {
              src: `/favicons/android-chrome-512x512.png`,
              sizes: `512x512`,
              type: `image/png`,
            },
          ],
          include_favicon: false,
          cache_busting_mode: false,
        })
        expect(headComponents).toMatchSnapshot()
      })
    })
  })

  describe(`Cache Busting`, () => {
    it(`doesn't add cache busting in manual mode`, () => {
      onRenderBody(ssrArgs, {
        icon: false,
        icons: [
          {
            src: `/favicons/android-chrome-48x48.png`,
            sizes: `48x48`,
            type: `image/png`,
          },
          {
            src: `/favicons/android-chrome-512x512.png`,
            sizes: `512x512`,
            type: `image/png`,
          },
        ],
        cache_busting_mode: `name`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`doesn't add cache busting if "cache_busting_mode" option is set to none`, () => {
      onRenderBody(ssrArgs, { icon: true, cache_busting_mode: `none` })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does file name cache busting if "cache_busting_mode" option is set to name`, () => {
      onRenderBody(ssrArgs, { icon: true, cache_busting_mode: `name` })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does query cache busting if "cache_busting_mode" option is set to query`, () => {
      onRenderBody(ssrArgs, { icon: true, cache_busting_mode: `query` })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does query cache busting if "cache_busting_mode" option is set to undefined`, () => {
      onRenderBody(ssrArgs, { icon: true })
      expect(headComponents).toMatchSnapshot()
    })
  })

  describe(`Favicon`, () => {
    it(`Adds link favicon tag if "include_favicon" is set to true`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        include_favicon: true,
        legacy: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does not add a link favicon if "include_favicon" option is set to false`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        include_favicon: false,
        legacy: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does not add a link favicon if in manual mode`, () => {
      onRenderBody(ssrArgs, {
        icon: false,
        icons: [
          {
            src: `/favicons/android-chrome-48x48.png`,
            sizes: `48x48`,
            type: `image/png`,
          },
          {
            src: `/favicons/android-chrome-512x512.png`,
            sizes: `512x512`,
            type: `image/png`,
          },
        ],
        legacy: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })
  })

  describe(`CORS Generation`, () => {
    it(`Adds a crossOrigin attribute to manifest link tag if provided`, () => {
      onRenderBody(ssrArgs, {
        crossOrigin: `use-credentials`,
        legacy: false,
        include_favicon: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Add crossOrigin when 'crossOrigin' is anonymous`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        crossOrigin: `anonymous`,
        legacy: false,
        include_favicon: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Does not add crossOrigin when 'crossOrigin' is blank`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        legacy: false,
        include_favicon: false,
        cache_busting_mode: `none`,
      })
      expect(headComponents).toMatchSnapshot()
    })
  })
})
