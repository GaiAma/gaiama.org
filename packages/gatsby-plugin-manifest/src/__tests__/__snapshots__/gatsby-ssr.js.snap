// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest CORS Generation Add crossOrigin when 'crossOrigin' is anonymous 1`] = `
Array [
  <link
    crossOrigin="anonymous"
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest CORS Generation Adds a crossOrigin attribute to manifest link tag if provided 1`] = `
Array [
  <link
    crossOrigin="use-credentials"
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest CORS Generation Does not add crossOrigin when 'crossOrigin' is blank 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Cache Busting Does file name cache busting if "cache_busting_mode" option is set to name 1`] = `
Array [
  <link
    href="/icons/icon-48x48-contentDigest.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48-contentDigest.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72-contentDigest.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96-contentDigest.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144-contentDigest.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192-contentDigest.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256-contentDigest.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384-contentDigest.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512-contentDigest.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Cache Busting Does query cache busting if "cache_busting_mode" option is set to query 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Cache Busting Does query cache busting if "cache_busting_mode" option is set to undefined 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Cache Busting doesn't add cache busting if "cache_busting_mode" option is set to none 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Cache Busting doesn't add cache busting in manual mode 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/favicons/android-chrome-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/favicons/android-chrome-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Favicon Adds link favicon tag if "include_favicon" is set to true 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Favicon Does not add a link favicon if "include_favicon" option is set to false 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Favicon Does not add a link favicon if in manual mode 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does create legacy links if "legacy" not specified in automatic mode 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does create legacy links if "legacy" not specified in hybrid mode. 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/favicons/android-chrome-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/favicons/android-chrome-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does create legacy links if "legacy" not specified in manual mode. 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/favicons/android-chrome-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/favicons/android-chrome-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does not create legacy links If "legacy" options is false and in automatic 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does not create legacy links If "legacy" options is false and in hybrid mode 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Legacy Icons Does not create legacy links If "legacy" options is false and in manual mode 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Add a "theme color" meta tag if "theme_color_in_head" is set to true 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=contentDigest"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Adds a "theme color" meta tag to head if "theme_color_in_head" is not provided 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Adds correct i18n "manifest" link to head 1`] = `
Array [
  <link
    href="/manifest_es.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Does not add a "theme color" meta tag if "theme_color_in_head" is set to false 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;

exports[`gatsby-plugin-manifest Manifest Link Generation Does not add a "theme_color" meta tag to head if "theme_color" option is not provided. 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
]
`;
